### ISSUE TITLE
About the Data class

### ISSUE THREAD

**Issue by Zawo1995** [conex-kge/ConEx#3]
Thanks for providing the source code for “ConvEx”.
As I search, I find that this code is from HyPER and TuckER.
I find that, in the original code, author set the _reverse_ param in Data is 'True' 
[TuckER](https://github.com/ibalazevic/TuckER/blob/23aff592115ee91e099ce490cdbea28ef026fc54/main.py#L194)
[HypER](https://github.com/ibalazevic/HypER/blob/1688b732047ad96cfae120748debb06a2fb7e209/HypER/hyper.py#L189)

but, you set the _reverse_ False except the **UMLS** dataset, could you explain why Doing such setting?

Looking forward to your response ;)

**Comment by conex-kge** [comment]
Dear @Zawo1995,

I thank you for your interest. Your observation is absolutely correct. The rationale behind of this is as follows:
Quotation from section 5.3
> ...After determining the best hyper-parameters based on the MRR on the validation dataset, we retrained ConEx again with the best found same hyper-parameters on the combination of train and valid sets that is commonly applied technique (see~\citep{joulin2017fast}).

To reproduce the reported results, you may need the best performing hyper parameters that are already stated in the supplemental material. Note that **r** stands for applying reciprocal learning.

```
\begin{table}[!htbp]
\centering
  \caption{Best performing hyperparameter values}
\begin{tabular}{lcccccc}
  \toprule
  Dataset & $\embdim$ & $dout_{\text{input}}$ & $dout_{\text{hidden}}$  & $dout_{\text{feature map}}$ & \text{channel-out}& r\\
  \midrule
  \approach-FB15K-237 & 20 & .1 & .1 & .1 & 64 &False \\
  \approach-WN18RR & 100 & .1 & .2 & .1 & 64&False\\
  \approach-KINSHIP & 100 & .3 & .5 & .3 & 2 &False\\
  \approach-UMLS & 100 & .4 & .4 & .4 &   8&True\\
  \midrule
  ComplEx-KINSHIP & 200 & .3 & - & -  &-&False\\
  ConvE-KINSHIP & 100 & .1 & .1 & .1 &  32 &False\\
  DistMult-KINSHIP & 200 & .3 & - & - &  - &False\\
  TuckER-KINSHIP & 100 & .3 & .4 \text{and} .5 & - &  - &True\\
  HypER-KINSHIP & 50 & .2 & .3 & .2 &  16 &False\\
  \midrule
  ComplEx-UMLS & 100 & .3 & - & - &-&False\\
  DistMult-UMLS & 100 & .3 & - & - &  - &False\\
  TuckER-UMLS & 200 & .3 & .4 \text{and} .5 & - &  - &True\\
  HypER-UMLS & 20 & .2 & .3 & .2 & 16 &False\\
  \bottomrule
\end{tabular}
    \label{table:hyperparams}
\end{table}
```

**Comment by Zawo1995** [comment]
Well, Kindly for replying so soon and providing what a important data in detail.
As we know, the _reverse_ in test_data is _must True_, because it means head prediction and tail prediction, if False, it means head prediction only, is that correct?
`self.test_data = self.load_data(data_dir, "test")`

**Comment by conex-kge** [comment]
Not including reverse triples on test split means predicting missing triples through tails. To investigate the impact of predicting missing triples through (only head), (only tail), (head and tail), we conducted further experiments on WN18, FB15K, WN18RR, FB15K-237 and YAGO3-10. We observed that there were no significant changes in the generalisation performance of ConEx on three variations.  These variations might have indeed have an impact of MRR as see issue #43 in ConvE.

We will update our manuscript and repo. by including (1) newly obtained results and (2) pretrained models soon.
